server/historian/packages/historian-base/src/services/restGitService.ts

```diff
@@ -460,15 +474,17 @@ export class RestGitService {
             }
 
             // Value is not cached - fetch it with the provided function and then cache the value
-            winston.info(`Fetching ${key}`);
-            Lumberjack.info(`Fetching ${key}`, this.lumberProperties);
-            const value = await fetch();
-            this.setCache(key, value);
-
-            return value;
-        } else {
-            return fetch();
+            return this.fetchAndCache(key, fetch);
+        }
+        if (resolvingSummary) {
```

Comment:
Looks like this `if` statement is independent from the top `if (this.cache && useCache)`. Imagine `this.cache` is define, and `useCache` is true. In case the summary cached value is found, you could be returning the `cachedValue` in line 473. And then, you would not have used `resolvingSummary` parameter. Was that intentional?

Reply:
Yes. :)

We only need to deal with the one corner case:
When we are making the first call to summary storage, we need to disable the cache, read the summary from storage, and update the cache.

The following calls will enable the cache, and return the correct result.

Reply:
Do you mean the first "GET summary" call? GET Summary happens after a POST summary, and when POSTing a summary, we also set it in the cache. https://github.com/microsoft/FluidFramework/blob/12d2622c05abe9be045c90bf938e7859d0a4d2b1/server/historian/packages/historian-base/src/services/restGitService.ts#L215

So what that means is that for the corner case you are mentioning, when the first GET summary happens, we would hit the cache since we had added the summary to the cache during the POST call. So even though you set `resolvingSummary` as true, your GET summary call would still be reading from the cache. Unless I'm missing something?

Reply:
Oh! Yes, good point! I should mention this:

In the driver side code, when we firstly make the call to get the summary, it should disable the cache. :)
https://github.com/microsoft/FluidFramework/pull/9706/files#:~:text=firstVersionsCall%20%26%26%20count%20%3D%3D%3D%201)%20%7B-,if%20(this.firstVersionsCall%20%26%26%20count%20%3D%3D%3D%201)%20%7B,const%20%7B%20id%3A%20_id%2C%20snapshotTree%20%7D%20%3D%20await%20this.fetchAndCacheSnapshotTree(latestSnapshotId%2C%20true)%3B,-return%20%5B%7B

Reply:
Is there a way that we can also do that here in this PR, without having to wait for the driver changes PR? In other words, since we cannot really control the driver version of clients talking to FRS - can this be a breaking change or cause undesired behavior if the client is not up-to-date?

Reply:
We cannot control the driver version of clients talking to FRS. So I have tested it with drivers with and without routing, and they all worked fine. :)

Meanwhile, if we used the driver without the discovery changes to test the driver with the discovery changes, the call to get the summary storage would use the same logic to cache and fetch logic.

Reply:
We cannot control the driver version of clients talking to FRS. So I have tested it with drivers with and without routing, and they all worked fine. :)

Meanwhile, if we used the driver without the discovery changes to test the driver with the discovery changes, the call to get the summary storage would use the same logic to cache and fetch logic.